android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/InteropTester.java

```diff
@@ -0,0 +1,354 @@
+      requests[i].payload = new Messages.Payload();
+    }
+    requests[0].payload.body = new byte[27182];
+    requests[1].payload.body = new byte[8];
+    requests[2].payload.body = new byte[1828];
+    requests[3].payload.body = new byte[45904];
+
+    final Messages.StreamingInputCallResponse goldenResponse =
+        new Messages.StreamingInputCallResponse();
+    goldenResponse.aggregatedPayloadSize = 74922;
+
+    StreamRecorder<Messages.StreamingInputCallResponse> responseObserver = StreamRecorder.create();
+    StreamObserver<Messages.StreamingInputCallRequest> requestObserver =
+        asyncStub.streamingInputCall(responseObserver);
+    for (Messages.StreamingInputCallRequest request : requests) {
+      requestObserver.onValue(request);
+    }
+    requestObserver.onCompleted();
+    assertEquals(goldenResponse, responseObserver.firstValue().get());
+  }
+
+  public void pingPong() throws Exception {
+    final Messages.StreamingOutputCallRequest[] requests =
+        new Messages.StreamingOutputCallRequest[4];
+    for (int i = 0; i < 4; i++) {
+      requests[i] = new Messages.StreamingOutputCallRequest();
+      requests[i].responseParameters = new Messages.ResponseParameters[1];
+      requests[i].responseParameters[0] = new Messages.ResponseParameters();
+      requests[i].payload = new Messages.Payload();
+    }
+    requests[0].responseParameters[0].size = 31415;
+    requests[0].payload.body = new byte[27182];
+    requests[1].responseParameters[0].size = 9;
+    requests[1].payload.body = new byte[8];
+    requests[2].responseParameters[0].size = 2653;
+    requests[2].payload.body = new byte[1828];
+    requests[3].responseParameters[0].size = 58979;
+    requests[3].payload.body = new byte[45904];
+
+
+    final Messages.StreamingOutputCallResponse[] goldenResponses =
+        new Messages.StreamingOutputCallResponse[4];
+    for (int i = 0; i < 4; i++) {
+      goldenResponses[i] = new Messages.StreamingOutputCallResponse();
+      goldenResponses[i].payload = new Messages.Payload();
+      goldenResponses[i].payload.type = Messages.COMPRESSABLE;
+    }
+    goldenResponses[0].payload.body = new byte[31415];
+    goldenResponses[1].payload.body = new byte[9];
+    goldenResponses[2].payload.body = new byte[2653];
+    goldenResponses[3].payload.body = new byte[58979];
+
+    final LinkedBlockingDeque<Messages.StreamingOutputCallResponse> responses =
+        new LinkedBlockingDeque();
+    final Messages.StreamingOutputCallResponse magicTailResponse =
+        new Messages.StreamingOutputCallResponse();
+    @SuppressWarnings("unchecked")
+    StreamObserver<Messages.StreamingOutputCallResponse> responseObserver =
+        new StreamObserver<Messages.StreamingOutputCallResponse>() {
+
+          @Override
+          public void onValue(Messages.StreamingOutputCallResponse value) {
+            responses.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
```

Comment:
The test will be failed if we don't get the response in 5 seconds, added a log so we know what was wrong in such case.